packages/shared/src/models/jsEvaluator/jsEvaluator.ts

```diff
@@ -23,6 +23,18 @@
 export class JSEvaluator {
   private evaluationContextBase: string;
 
+  private readonly restrictedIdentifiers = [
+    "window",
+    "name",
+    "globalThis",
+    "self",
+    "document",
+    "parent",
+    "top",
+    "frames",
+    "global",
+  ];
```

Comment:
nit(blocking): I think that we currently potentially make use of some of these (specifically "window" and "document"). So probably safest to just restrict "name" and "parent" for now (as these are likely to be the ones that could conflict with variable values)? Perhaps it would make sense to leave the other strings here in the list but comment them out, as it could be useful reference in future.